crates/typst/src/symbols/symbol.rs

```diff
@@ -46,47 +46,63 @@ pub use typst_macros::symbols;
 #[derive(Debug, Clone, Eq, PartialEq, Hash)]
 pub struct Symbol(Repr);
 
+/// The character of a symbol, possibly with a function.
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub enum SymChar {
```

Comment:
I think a struct with fields of type `char` and `Option<fn() -> Func>` would be a bit simpler.